CMakeLists.txt

```diff
@@ -53,11 +53,15 @@ add_subdirectory("log4tango")
 add_subdirectory("cppapi")
 
 if(BUILD_TESTING)
-  if(BUILD_SHARED_LIBS)
+  find_program(DOCKER_BINARY docker)
+
+  if(NOT DOCKER_BINARY)
+      message(WARNING "The tests can not be run as docker is missing.")
+  elseif(BUILD_SHARED_LIBS)
     add_subdirectory("cpp_test_suite")
   else()
     message(WARNING "Not building the tests, because that is currently supported only when BUILD_SHARED_LIBS is ON")
-    SET(BUILD_TESTING OFF)
+    set(BUILD_TESTING OFF)
   endif()
 endif()
```

Comment:
I think these if-elseif-else are a bit hard to read, partly because the 'usual' execution path is in the middle elseif. I'd print both warning messages even if we don't have docker and if we are doing static build because these two things are really unrelated.

Also note that currently if we don't have docker and are doing a static build build, we will not clear the BUILD_TESTING option, which may be a problem if we attempt to check this option later or in some other cmake file.

I suggest to do a minimal change. Either check for docker binary just after `add_subdirectory(...)`:
```cmake
if (BUILD_TESTING)
  if (BUILD_SHARED_LIBS)
    add_subdirectory("cpp_test_suite")

    # added
    find_program(DOCKER_BINARY docker)
    if (NOT DOCKER_BINARY)
      message()
    endif()
  endif()
endif()
```
Or if you don't like double-nesting, after we check for static build:
```cmake
if (BUILD_TESTING)
  if (BUILD_SHARED_LIBS)
    add_subdirectory("cpp_test_suite")
  else()
    set(BUILD_TESTING OFF)
  endif()
endif()

# added
if (BUILD_TESTING)
  find_program(DOCKER_BINARY docker)
  if (NOT DOCKER_BINARY)
    message()
  endif()
endif()
```

Reply:
Actually docker check can go straight into the first BUILD_TESTING check. User intention was to run the tests so I think we can keep the warning even if we disabled tests for any other reason.

```cmake
if (BUILD_TESTING)
  # added
  find_program(DOCKER_BINARY docker)
  if (NOT DOCKER_BINARY)
    message()
  endif()

  if (BUILD_SHARED_LIBS)
    add_subdirectory("cpp_test_suite")
  else()
    set(BUILD_TESTING OFF)
  endif()
endif()
```

Reply:
@mliszcz Jeep. I've used the last suggestion. Thanks.